spec/linkedlist.js

```diff
@@ -0,0 +1,339 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import LinkedList from '../src/linkedlist'
+
+chai.use(chaiChange)
+
+describe('LinkedList', () => {
+  'use strict'
+
+  it('exists', () => {
+    expect(LinkedList).to.be.a('function')
+  })
+
+  context('getHeadNode()', () => {
+    it('Returns the first node in the list', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Mercedes")
+      myLinkedList.insertFirst("Lexus")
+
+      expect(myLinkedList.getHeadNode().data).to.equal("Lexus")
+    })
+  })
+
+  context('getHeadNode()', () => {
+    it('Returns the first node in the list', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Mercedes")
+
+      expect(myLinkedList.getHeadNode().data).to.equal("BMW")
+    })
+  })
+
+  context('getTailNode()', () => {
+    it('Returns the last node in the list', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Mercedes")
+      myLinkedList.insertFirst("Lexus")
+
+      expect(myLinkedList.getTailNode().data).to.equal("Mercedes")
+    })
+  })
+
+  context('getTailNode()', () => {
+    it('Returns the last node in the list', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+
+      expect(myLinkedList.getTailNode().data).to.equal("BMW")
+    })
+  })
+
+  context('insert()', () => {
+    it('Inserts a node (with the provided data) to an empty linked list, and sets the head to equal the tail ', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      expect(myLinkedList.head).to.equal(myLinkedList.tail)
+    })
+  })
+
+  context('insert()', () => {
+    it('Inserts multiple nodes (with the provided data) to an empty linked list, and checks the value of the head ', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      expect(myLinkedList.head.data).to.equal("BMW")
+    })
+  })
+
+  context('insert()', () => {
+    it('Inserts multiple nodes (with the provided data) to an empty linked list, and checks the value of the head ', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
```

Comment:
so many of these tests have the same context, and the same 3 lines in them. To keep a good economic code file I would suggest just having all of your different expects in one context function, this way it is more readable for the reviewer and for you in the future.